adapters/mobkoi/mobkoi.go

```
@@ -0,0 +1,108 @@
+package mobkoi
+
+import (
+	"net/http"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type BidderExt struct {
+	Bidder openrtb_ext.ImpExtMobkoi `json:"bidder"`
+}
+
+type UserExt struct {
+	Consent string `json:"consent"`
+}
+
+// Builder builds a new instance of the {bidder} adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	ext := BidderExt{}
+	if err := jsonutil.Unmarshal(request.Imp[0].Ext, &ext); err != nil {
+		return nil, []error{err}
+	}
+
+	if request.Imp[0].TagID == "" {
+		request.Imp[0].TagID = ext.Bidder.PlacementID
+	}
+
+	uri := a.endpoint
+	if ext.Bidder.AdServerBaseUrl != "" {
```

Do we need to validate this URL to avoid the chances of invalid URL in that case configured endpoint (a.endpoint) can be used? 

Suggestion:
```
if ext.Bidder.AdServerBaseUrl != "" {
    baseURL, err := url.Parse(ext.Bidder.AdServerBaseUrl)
    if err == nil { // Ensure parsing doesn't fail
        baseURL.Path = strings.TrimRight(baseURL.Path, "/") + "/bid"
        uri = baseURL.String()
    }
}

```

---

I don't find it necessary, because it will fail silently and hide the issue.

---

The intent was to use the configured endpoint (a.endpoint) if the URL coming in request is invalid

---

I added the fallback as you suggested. I use `ParseRequestURI` though, `Parse` is not a reliable validator